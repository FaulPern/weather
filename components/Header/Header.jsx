import styles from './Header.module.css'
import Link from 'next/link'

const HeaderLink = ({ href, as, children }) => {
  const { listItem } = styles

  return (
    <li className={listItem}>
      <Link href={href} as={as}>
        <a>{children}</a>
      </Link>
    </li>
  )
}

const Header = () => {
  const { header, list } = styles
  const postPageLine = '/post/[id]'
  return (
    <header className={header}>
      <ul className={list}>
        <HeaderLink href='/index' as='/'>Home</HeaderLink>
        <HeaderLink as='/post/1' href={postPageLine}>post 1</HeaderLink>
        <HeaderLink as='/post/2' href={postPageLine}>post 2</HeaderLink>
        <HeaderLink as='/post/3' href={postPageLine}>404</HeaderLink>
      </ul>
    </header>
  )
}

export default Header
