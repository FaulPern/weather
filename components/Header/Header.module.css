.header {
    background: #eee;
    border-bottom: 1px solid #aaa;
    padding: 1rem 0rem;
    font-size: 1.2rem;
}

.list {
    padding: 0;
    margin: 0;
    display: flex;
}

.listItem {
    list-style: none;
    padding: 0 .5rem;
}