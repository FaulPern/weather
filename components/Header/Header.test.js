import { shallow, mount } from 'enzyme'
import Header from './Header.jsx'

describe('Header component', () => {
  const mountComponent = mount(<Header />)
  const shallowComponent = shallow(<Header />)
  it('should have 4 link element with direct "a" tag', () => {
    expect(mountComponent.find('Link > a')).toHaveLength(4)
  })

  it('should have 4 HeaderLink component', () => {
    expect(shallowComponent.find('Link > a')).toHaveLength(0) // shallow will not render child component
    expect(shallowComponent.find('HeaderLink')).toHaveLength(4)
  })

  it('should render', () => {
    expect(mountComponent).toMatchSnapshot()
  })
})
