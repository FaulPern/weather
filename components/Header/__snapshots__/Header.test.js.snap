// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
<Header>
  <header
    className="header"
  >
    <ul
      className="list"
    >
      <HeaderLink
        as="/"
        href="/index"
      >
        <li
          className="listItem"
        >
          <Link
            as="/"
            href="/index"
          >
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Home
            </a>
          </Link>
        </li>
      </HeaderLink>
      <HeaderLink
        as="/post/1"
        href="/post/[id]"
      >
        <li
          className="listItem"
        >
          <Link
            as="/post/1"
            href="/post/[id]"
          >
            <a
              href="/post/1"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              post 1
            </a>
          </Link>
        </li>
      </HeaderLink>
      <HeaderLink
        as="/post/2"
        href="/post/[id]"
      >
        <li
          className="listItem"
        >
          <Link
            as="/post/2"
            href="/post/[id]"
          >
            <a
              href="/post/2"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              post 2
            </a>
          </Link>
        </li>
      </HeaderLink>
      <HeaderLink
        as="/post/3"
        href="/post/[id]"
      >
        <li
          className="listItem"
        >
          <Link
            as="/post/3"
            href="/post/[id]"
          >
            <a
              href="/post/3"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              404
            </a>
          </Link>
        </li>
      </HeaderLink>
    </ul>
  </header>
</Header>
`;
